pkg/rbd/rbd_util_test.go

```diff
@@ -54,3 +54,80 @@ func TestHasSnapshotFeature(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateAndGetImageFeaturesWithMounter(t *testing.T) {
+	tests := []struct {
+		volOptions     map[string]string
+		isImageFeature string
+		isMounter      string
+		isErrMsg       string
+	}{
+		{
+			map[string]string{"imageFeatures": "layering", "mounter": rbdDefaultMounter},
+			"layering",
+			rbdDefaultMounter,
+			"",
+		},
+		{
+			map[string]string{"imageFeatures": "layering,exclusive-lock", "mounter": rbdNbdMounter},
+			"layering,exclusive-lock",
+			rbdNbdMounter,
+			"",
+		},
+		{
+			map[string]string{"imageFeatures": "layering,exclusive-lock,journaling", "mounter": rbdNbdMounter},
+			"layering,exclusive-lock,journaling",
+			rbdNbdMounter,
+			"",
+		},
+		{
+			map[string]string{"imageFeatures": "layering,journaling", "mounter": rbdNbdMounter},
+			"",
+			"",
+			"feature journaling requires exclusive-lock",
+		},
+		{
+			map[string]string{"imageFeatures": "layering,exclusive-lock,journaling"},
+			"",
+			"",
+			"feature exclusive-lock requires rbd-nbd for mounter",
+		},
+		{
+			map[string]string{"imageFeatures": "layering,exclusive-lock,journaling", "mounter": rbdDefaultMounter},
+			"",
+			"",
+			"feature exclusive-lock requires rbd-nbd for mounter",
+		},
+		{
+			map[string]string{"imageFeatures": "layering,exclusive-loc,journaling", "mounter": rbdNbdMounter},
+			"",
+			"",
+			"invalid feature exclusive-loc for volume csi-rbdplugin",
+		},
+		{
+			map[string]string{"imageFeatures": "ayering", "mounter": rbdDefaultMounter},
+			"",
+			"",
+			"invalid feature ayering for volume csi-rbdplugin",
+		},
+	}
+
+	for _, test := range tests {
+		imageFeatures, mounter, err := validateAndGetImageFeaturesWithMounter(test.volOptions)
+
+		if !isCorrectError(err, test.isErrMsg) || (imageFeatures != test.isImageFeature) || (mounter != test.isMounter) {
+			t.Errorf("validateAndGetImageFeaturesWithMounter(%#v) = %s/%s/%v, want %s/%s/%s",
+				test.volOptions, imageFeatures, mounter, err, test.isImageFeature, test.isMounter, test.isErrMsg)
+		}
+	}
+}
+
+func isCorrectError(err error, errorMsg string) bool {
```

Comment:
Instead of hand-rolling functions like this perhaps it would be better to start using a packge like testify?

Even if not, treating nil errors as equal to an empty string seems like a trap for future devs. If you keep this function I highly recommend rolling a nil-check and a same-content check as two different functions.

Reply:
yes, that's correct. @Madhu-1 Do we have a package to check for errors? If not, I can add an appropriate package, or split the function as suggested by @phlogistonjohn

Reply:
currently, we don't have any package, but you can check https://godoc.org/github.com/stretchr/testify

Reply:
Hi @woohhan, are you planning to update this PR with the suggestion?

Reply:
just now I updated the PR with the suggestion!